gpytorch/lazy/lazy_tensor.py

```diff
@@ -424,7 +424,7 @@ def _cholesky_solve(self, rhs):
         Returns:
             (LazyTensor) Cholesky factor
         """
-        return torch.cholesky_solve(rhs, self.evaluate())
+        return torch.cholesky_solve(rhs, self.evaluate(), upper=upper)
```

Comment:
Now that we have `TriangularLazyTensor`, this method could move to `TriangularLazyTensor` and throw an error for the base LazyTensor.

Reply:
makes sense. What I will do is raise a `NotImplementedError` here. Then in `TriangularLazyTensor._cholesky_solve` I will try calling the `_cholesky_solve` method of `_tensor` and catch the `NotImplementedError` and use the `TriangularLazyTensor` method. That way we can still use the custom implementations of all the other lazy tensors.